### PR TITLE
Fix professor login

### PR DESCRIPTION
**Pull request by NEUDitao**

### CODE REVIEW

File: packages/server/src/login/login-course.service.ts [hunk 1]
```diff
@@ -51,23 +46,21 @@ export class LoginCourseService {
...
-          );
-          userCourses.push(taCourse);
-        }
-      }),
-    );
+      for (const courseMapping of courseMappings) {
+        const taCourse = await this.courseToUserCourse(
+          user.id,
+          courseMapping.courseId,
+          c.instructor === 1 ? Role.PROFESSOR : Role.TA,
```
Comment (liustanley) [inline packages/server/src/login/login-course.service.ts:59]: Can we also change the type in `KhouryTACourse` to `number`?

File: packages/server/test/login.integration.ts [hunk 2]
```diff
@@ -286,7 +286,7 @@ describe('Login Integration', () => {
       expect(ta.courses).toHaveLength(3);
     });
 
-    it('handles professor courses correctly', async () => {
+    it.only('handles professor courses correctly', async () => {
```
Comment (liustanley) [inline packages/server/test/login.integration.ts:289]: Remove `.only`

**Review by liustanley — CHANGES_REQUESTED** [review]

File: packages/server/src/login/login-course.service.ts (hunk 1 above)
Comment (liustanley) [reply packages/server/src/login/login-course.service.ts:59]: Just checked and cypress is failing because of this too

Comment (NEUDitao) [reply packages/server/src/login/login-course.service.ts:59]: wait lol I just forgot to cmmit it HAHA

File: packages/server/test/login.integration.ts (hunk 2 above)
Comment (NEUDitao) [reply packages/server/test/login.integration.ts]: yes sir

**Review by willstenzel — APPROVED** [review]

**Review by liustanley — APPROVED** [review]

**Pull request merged**